src/licensedcode/models.py

```diff
@@ -658,6 +662,12 @@ def validate(licenses, verbose=False, no_dupe_urls=False):
             # local text consistency
             text = lic.text
 
+            data = {"text": text}
+            # We are testing whether we can dump as yaml and load from yaml
+            # without failing (i.e. whether the text is yaml safe) 
+            yaml_string = yaml_dump(data, indent=4)
+            loaded_yaml = yaml_load(yaml_string)
```

Comment:
1. If we test also for load() with plain library we are getting [scancode.lockfile.LockTimeout error](https://dev.azure.com/nexB/scancode-toolkit/_build/results?buildId=10460&view=logs&j=56739ecf-8d40-52c7-9da4-d79b306f05c9&t=c0917742-18c4-555f-7c19-b4b94cb323a4&l=13642) but the tests pass so this works allright.
2. validate that the reloaded yaml is the same as the data in both cases: this fails because `saneyaml.dump()` strips adds/newlines, I looked a bit into saneyaml but it is not obvious how to fix this.
3. capture the exception and report these as errors: Done!

Reply:
Fixed these by moving the validate outside the index file lock, and also making sure we don't run any of the `thorough` load/dump checks while creating the index. But perform these expensive operations only on the test side.

Reply:
Tracking the point 2. here in another issue separately as this is a saneyaml issue: https://github.com/nexB/saneyaml/issues/10